### DISCUSSION TITLE
Barcode scanner widget

### DISCUSSION THREAD

**Discussion by mrgrimshaw** [opengisch/QField#3841] [Ideas]
Hi,

It would be great if one of the collection fields could be set up so that when entering text you can scan a barcode. When collecting samples i find manually typing sample ID results in errors and we can get tickets that have barcodes. Apologies if this is already possible (i'd love to know).

Thanks

**Comment by nirvn** [comment]
@mrgrimshaw , in the latest version of QField, you'll find a new menu attached to the _text edit widget_ within the feature form. In that menu, a 'Scan Code' action will do exactly what you are looking for:
![image](https://user-images.githubusercontent.com/1728657/214809140-cb1f76af-31d8-4cc0-a5ee-f8f539e4437e.png)

**Reply by mrgrimshaw** [reply]
Legend, really useful. Just have to find a reliable seller of barcode sample ticket books now.
